Fibonacci

freakyblue opened the pull request.

freakyblue committed:
Add fibonacci task

freakyblue committed:
Add exception test

freakyblue committed:
Rename folder

freakyblue committed:
Right method calling in test

freakyblue committed:
Remove old fibonacci folder

freakyblue committed:
Removed (second) headline, add link to tests

Task title show automatic already.

freakyblue committed:
Remove (second) headline, add link to tests

Task title show automatic already.

freakyblue committed:
Merge branch 'fibonacci' of https://github.com/st-tu-dresden/inloop-sample-tasks into fibonacci

martinmo commented on .gitignore:
```diff
@@ -0,0 +1 @@
+.class
```
`*.class`

martinmo commented on Fibonacci/meta.json:
```diff
@@ -0,0 +1,6 @@
+{
+    "author": "Software Technology Group, TU Dresden",
+    "category": "Basic",
+    "pubdate": "2017-05-03 00:42:42",
```
Please use a more meaningful pubdate ;-)

freakyblue replied:
Which one should I use?

martinmo commented on Fibonacci/solution/Fibonacci.java:
```diff
@@ -0,0 +1,17 @@
+
+  public int calcFib (int n) {
+    if (n < 0)
+      throw new IllegalArgumentException("Fibonacci is not defined for neagtive Numbers!");
```
neagtive => negative

martinmo commented on Fibonacci/task.md:
```diff
@@ -0,0 +1,8 @@
+Write a class called `Fibonacci` within the methode `calcFib (int n)` which returns
+the result of the Fibonacci function of given number `n`.
+
+Make sure that only legal arguments will be accepted otherwise throw an Exception `Fibonacci is not defined for neagtive Numbers!`.
```
neagtive => negative
Exception => IllegalArgumentException

martinmo commented on Fibonacci/solution/Fibonacci.java:
```diff
@@ -0,0 +1,17 @@
+class Fibonacci {
```
`public class`

martinmo reviewed (changes requested):
Thanks for your work, Jonny! Here are my comments:

The `Fibonacci` class: I think `calcFib()` should be static and the name changed to `fibonacci()`.

The task description is full of typos and generally must be improved: what exactly is meant by *n*? You should explain that you want the method to compute the *n-th element* of the *fibonacci sequence*. While you are at this, you should mention whether your test expects the sequence to start with 0 and 1 or 1 and 1 (there are two "versions" of fib, the first one is the "modern" one, see https://en.wikipedia.org/wiki/Fibonacci_number).

Please don't include binary *.zip files in the repo and adapt the coding style of the model solution.

freakyblue committed:
Include critizied changes.

Rename calcFib to fibonacci, make it static, fix typos,
add example to task description.

martinmo commented on Fibonacci/task.md:
```diff
@@ -1,7 +1,20 @@
+
+## Example ##
+
+Here are some values to show, which results the function `fibonacci (n)` should return.
```
Please remove the unnecessary comma after "show" and end the sentence with a ":".

martinmo commented on Fibonacci/task.md:
```diff
@@ -1,7 +1,20 @@
+
+Here are some values to show, which results the function `fibonacci (n)` should return.
+
+`n` --> `fibonacci (n)`  
```
Good idea! But why don't you use a Markdown table (see https://guides.github.com/features/mastering-markdown/)?

freakyblue replied:
I did it first, but INLOOP don't present it as as table. I think the [Leetspeak-task](https://github.com/st-tu-dresden/inloop-tasks/blob/master/LeetSpeek/task.md) has the same problem and solved it with a HTML table, but the default spacing between the columns is too close, in my opinion. So the arrows seemed to be a fair and clean way.

martinmo replied:
Ok. Good point! I recall that there was a limitation in the Markdown library that we use in INLOOP.

martinmo commented on Fibonacci/task.md:
```diff
@@ -0,0 +1,21 @@
+Write a class called `Fibonacci` within the function `fibonacci (int n)` which returns
+the result of the Fibonacci function of given number `n`.
```
can you please change this to:
> Implement a class `Fibonacci` containing a static method `fibonacci(int n)`, which returns the n-th element of the fibonacci sequence.

martinmo commented on Fibonacci/task.md:
```diff
@@ -0,0 +1,21 @@
+Write a class called `Fibonacci` within the function `fibonacci (int n)` which returns
+the result of the Fibonacci function of given number `n`.
+
+Make sure that only legal arguments will be accepted otherwise throw an
```
```
accepted otherwise
        ^-- insert comma here
```

martinmo reviewed (commented):
Thanks for the update :+1:

After you have addressed the comments, you will need to complete the Dockerfile and create a entrypoint script.

freakyblue committed:
Fixed typos in task.md

freakyblue committed:
Add entrypoint.sh

Just copied the file from the original repo.

freakyblue commented:
Is there anything missing? Or can the branch be merged?

martinmo commented:
Yes, the `Dockerfile` and `Makefile` are empty! And ideally you should also add a `.dockerignore` file with according entries to exclude the model solutions from the resulting Docker image. You can get an idea of what needs to be in these files by looking at our private repository.

martinmo commented:
And I've spotted two other small typos: *Number* is incorrectly capitalized in the solution and task description.

freakyblue committed:
Fix typo
